modules/adpod.js

```diff
@@ -0,0 +1,356 @@
+    attachPriceIndustryDurationKeyToBid(bidResponse);
+
+    updateBidQueue(auctionInstance, bidResponse, afterBidAdded);
+  } else {
+    fn.call(this, auctionInstance, bidResponse, afterBidAdded, bidderRequest);
+  }
+}
+
+/**
+ * This hook function will review the adUnit setup and verify certain required are present in any adpod adUnits.
+ * If the fields are missing or incorrectly setup, the adUnit is removed from the list.
+ * @param {Function} fn reference to original function (used by hook logic)
+ * @param {Array[Object]} adUnits list of adUnits to be evaluated
+ * @returns {Array[Object]} list of adUnits that passed the check
+ */
+export function checkAdUnitSetupHook(fn, adUnits) {
+  let goodAdUnits = adUnits.filter(adUnit => {
+    let mediaTypes = utils.deepAccess(adUnit, 'mediaTypes');
+    let videoConfig = utils.deepAccess(mediaTypes, 'video');
+    if (videoConfig && videoConfig.context === ADPOD) {
+      // run check to see if other mediaTypes are defined (ie multi-format); reject adUnit if so
+      if (Object.keys(mediaTypes).length > 1) {
+        utils.logWarn(`Detected more than one mediaType in adUnitCode: ${adUnit.code} while attempting to define an 'adpod' video adUnit.  'adpod' adUnits cannot be mixed with other mediaTypes.  This adUnit will be removed from the auction.`);
+        return false;
+      }
+
+      let errMsg = `Detected missing or incorrectly setup fields for an adpod adUnit.  Please review the following fields of adUnitCode: ${adUnit.code}.  This adUnit will be removed from the auction.`;
+
+      let playerSize = !!(videoConfig.playerSize && utils.isArrayOfNums(videoConfig.playerSize));
+      let adPodDurationSec = !!(videoConfig.adPodDurationSec && utils.isNumber(videoConfig.adPodDurationSec));
+      let durationRangeSec = !!(videoConfig.durationRangeSec && utils.isArrayOfNums(videoConfig.durationRangeSec));
+
+      if (!playerSize || !adPodDurationSec || !durationRangeSec) {
+        errMsg += (!playerSize) ? '\nmediaTypes.video.playerSize' : '';
+        errMsg += (!adPodDurationSec) ? '\nmediaTypes.video.adPodDurationSec' : '';
+        errMsg += (!durationRangeSec) ? '\nmediaTypes.video.durationRangeSec' : '';
+        utils.logWarn(errMsg);
+        return false;
+      }
+    }
+    return true;
+  });
+  adUnits = goodAdUnits;
+  fn.call(this, adUnits);
+}
+
+/**
+ * This check evaluates the incoming bid's `video.durationSeconds` field and tests it against specific logic depending on adUnit config.  Summary of logic below:
+ * when adUnit.mediaTypes.video.requireExactDuration is true
+ *  - only bids that exactly match those listed values are accepted (don't round at all).
+ *  - populate the `bid.video.durationBucket` field with the matching duration value
+ * when adUnit.mediaTypes.video.requireExactDuration is false
+ *  - round the duration to the next highest specified duration value based on adunit (eg if range was [5, 15, 30] -> 3s is rounded to 5s; 16s is rounded to 30s)
```

Comment:
we wanted to add a 2 second buffer here. That is if a bid is 16s it should be rounded down to the the nearest bucket (15s in this case) because it is within 2s of a bucket.

Reply:
I'll make an update here to incorporate this logic.  Should this buffer apply to the highest range?  ie if the ranges were `[15, 30, 45, 60]` and the bid was `61` - we would round it down to 60 and accept it?

Reply:
yeah I think we can apply it to the highest value as well.

Reply:
I updated the logic to account for this buffer; please take another look when you have the chance.